src/room/GroupCallView.tsx

```diff
@@ -187,24 +179,31 @@ export const GroupCallView: FC<Props> = ({
 
     if (widget && preload && skipLobby) {
       // In preload mode without lobby we wait for a join action before entering
-      const onJoin = async (
-        ev: CustomEvent<IWidgetApiRequest>,
-      ): Promise<void> => {
-        await defaultDeviceSetup(ev.detail.data as unknown as JoinCallData);
-        await enterRTCSession(rtcSession, perParticipantE2EE);
-        await widget!.api.transport.reply(ev.detail, {});
+      const onJoin = (ev: CustomEvent<IWidgetApiRequest>): void => {
+        defaultDeviceSetup(ev.detail.data as unknown as JoinCallData)
+          .catch((e) => {
+            logger.error("Error setting up default devices", e);
+          })
+          .then(async () => enterRTCSession(rtcSession, perParticipantE2EE))
+          .then(() => widget!.api.transport.reply(ev.detail, {}))
+          .catch((e) => {
+            logger.error("Error entering RTC session", e);
+          });
```

Comment:
The first `.then` callback (and its follow-ups) will be reached even if `defaultDeviceSetup` rejects, which I assume is not the intention here, given that the original implementation would early-exit if any of the called methods were to throw.

An easy way to get the same early-exit behaviour as the original while still returning `void` (at the cost of less granular error logging) is to wrap the `await`ed calls in an anonymous async function, which is a pattern I think I've seen elsewhere in this codebase:
```suggestion
        (async (): Promise<void> => {
          await defaultDeviceSetup(ev.detail.data as unknown as JoinCallData);
          await enterRTCSession(rtcSession, perParticipantE2EE);
          widget!.api.transport.reply(ev.detail, {});
        })().catch((e) => {
          logger.error("Error joining RTC session", e);
        });
```

The same can be done a few lines down.